libraries/lib-datahandler/src/main/thrift/components.thrift

```diff
@@ -206,6 +206,7 @@ struct Release {
     53: optional set<string> operatingSystems,
     54: optional COTSDetails cotsDetails,
     55: optional EccInformation eccInformation,
+    56: optional set<string> softwarePlatforms,
```

Comment:
This new field is completely independent from the list of software platforms at the component level? Why is it needed?

Reply:
I don't understand the first question. It is needed because we report on the release level and not on the component level.

Reply:
There is already a field "softwarePlatforms" at the component level. As a release always belongs to a component I was wondering if the fields should really be independent. Or if there should be some logic like "the list of software platforms of a component is the aggregation of software platform lists of its releases".
Beside that your core idea here seems to be that the content of software platforms can change between different releases of the same component. Is that correct? Otherwise you could just get the information from the component of the release even when you report on release level.

Reply:
> There is already a field "softwarePlatforms" at the component level. As a release always belongs to a component I was wondering if the fields should really be independent. Or if there should be some logic like "the list of software platforms of a component is the aggregation of software platform lists of its releases" now that we have this field at release level.

Since a release already has it's own "operating systems" & "programming languages" i assumed that "software platforms" should follow suit. If this is wrong, what is the purpose of "operating systems" & "programming languages" on the release level?

> Beside that your core idea here seems to be that the content of software platforms can change between different releases of the same component. Is that correct? Otherwise you could just get the information from the component of the release even when you report on release level.

Yes, that is correct. Isn't the component suppose to be more of an abstract/template of a release and the each release is what actually goes into the project?

Reply:
maybe the field softwarePlatforms should not been created at the first hand, but if in the report, it should be taken from the component where the release belongs to. I think a similar case is type (OSS, COTS, Freeware, ...) where the field value is taken from the component when listing releases.